### PR TITLE
feature: Add an interface for confirming file download

### PR DESCRIPTION
**Pull request by HoonBaek**
[CLNP-2565](https://sendbird.atlassian.net/browse/CLNP-2565)

### Original Reason
* There was no interface available to decide whether to proceed with the file download or not

### ChangeLog & Feature
* Added `onBeforeDownloadFileMessage` to the `<GroupChannel />` and `<Thread />` modules
* Added `onDownloadClick` to the `FileViewer`, `FileViewerView`, `MobileBottomSheet`, `MobileContextMenu`, and `MobileMenu`

### How to use
```tsx
const ONE_MB = 1024 * 1024;
/**
  * Use this list to check if it's displayed as a ThumbnailMessage.
  * (https://github.com/sendbird/sendbird-uikit-react/blob/main/src/utils/index.ts)
*/
const ThumbnailMessageTypes = [
  'image/jpeg',
  'image/jpg',
  'image/png',
  'image/gif',
  'image/svg+xml',
  'image/webp', // not supported in IE
  'video/mpeg',
  'video/ogg',
  'video/webm',
  'video/mp4',
];

<GroupChannel // or Thread
  onBeforeDownloadFileMessage={async ({ message, index = null }) => {
    if (message.isFileMessage()) {
      const confirmed = window.confirm(`The file size is ${(message.size / ONE_MB).toFixed(2)}MB. Would you like to continue downloading?`);
      return confirmed;
    }
    if (message.isMultipleFilesMessage()) {
      const confirmed = window.confirm(`The file size is ${(message.fileInfoList[index].fileSize / ONE_MB).toFixed(2)}MB. Would you like to continue downloading?`);
      return confirmed;
    }
    return true;
  }}
/>
```

[CLNP-2565]: https://sendbird.atlassian.net/browse/CLNP-2565?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/modules/GroupChannel/components/Message/MessageView.tsx [hunk 1]
```diff
@@ -80,6 +80,11 @@ export interface MessageViewProps extends MessageProps {
 
   renderFileViewer: (props: { message: FileMessage; onCancel: () => void }) => React.ReactElement;
   renderRemoveMessageModal?: (props: { message: EveryMessage; onCancel: () => void }) => React.ReactElement;
+  /**
+   * You can't use this prop in the (legacy) Channel component.
```
Comment (liamcho) [inline src/modules/GroupChannel/components/Message/MessageView.tsx:84]:
```suggestion
   * You can't use this prop in the Channel component (legacy).
```

File: src/modules/Thread/components/ParentMessageInfo/ParentMessageInfoItem.tsx [hunk 2]
```diff
@@ -93,6 +97,29 @@ export default function ParentMessageInfoItem({
     });
   }, [message?.updatedAt, (message as UserMessage)?.message]);
 
+  // Only for the FileMessageItemBody
+  const downloadFileWithUrl = () => {
+    if (message.messageType === 'file') {
```
Comment (liamcho) [inline src/modules/Thread/components/ParentMessageInfo/ParentMessageInfoItem.tsx:102]: what about multiple files?

File: src/modules/Thread/components/ThreadList/ThreadListItem.tsx [hunk 3]
```diff
@@ -258,13 +259,27 @@ export default function ThreadListItem({
...
+            }
+            try {
+              const allowDownload = await onBeforeDownloadFileMessage({ message: message as FileMessage | MultipleFilesMessage });
+              if (!allowDownload) {
+                e.preventDefault();
+                logger.info?.('ThreadListItem: Not allowed to download.');
+              }
+            } catch (err) {
+              logger.error?.('ThreadListItem: Error occurred while determining download continuation:', err);
+            }
```
Comment (liamcho) [inline src/modules/Thread/components/ThreadList/ThreadListItem.tsx:281]: maybe we can take this repeated code out to utils and use import intead?

**Review by liamcho — CHANGES_REQUESTED** [review]
please see my comment

File: src/modules/Thread/components/ParentMessageInfo/ParentMessageInfoItem.tsx (hunk 2 above)
Comment (HoonBaek) [reply src/modules/Thread/components/ParentMessageInfo/ParentMessageInfoItem.tsx:102]: We download files on the Fileviewer component for the MFM, not using the link

File: src/modules/Thread/components/ThreadList/ThreadListItem.tsx (hunk 3 above)
Comment (HoonBaek) [reply src/modules/Thread/components/ThreadList/ThreadListItem.tsx:281]: yes, but let me do that later, we should merge quickly ;; 😥

**Review by liamcho — APPROVED** [review]
LGTM

**Pull request merged**